mesonbuild/environment.py

```diff
@@ -149,6 +149,9 @@ def find_coverage_tools(coredata: coredata.CoreData) -> T.Tuple[T.Optional[str],
     gcovr_exe, gcovr_version = detect_gcovr()
 
     llvm_cov_exe = detect_llvm_cov(compute_llvm_suffix(coredata))
+    # Some platforms may provide versioned clang but only non-versioned llvm utils
+    if llvm_cov_exe is None:
+        llvm_cov_exe = detect_llvm_cov('')
```

Comment:
Maybe the actual fix is that detect_llvm_cov should first try with a suffix and fall back to trying without one? Currently if you pass an explicit suffix it skips auto detection entirely via if/else -- in this case I'd probably make it a new else clause for shutil.which

Reply:
I thought about that too, but then providing an explicit version to `detect_llvm_cov` and getting back a llvm-cov of potentially different version seems bad.

Reply:
Fair.